eclipse-che/che-operator
# feat: CheCluster defaults

Signed-off-by: Anatolii Bazko <abazko@redhat.com>



### What does this PR do?
Set CheCluster defaults

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20804

### How to test this PR?


```
oc create namespace eclipse-che
oc apply -f config/crd/bases/org.eclipse.che_checlusters.yaml
oc apply -f config/samples/org_v2_checluster.yaml -n eclipse-che 
oc get checluster/eclipse-che -o yaml -n eclipse-che
```

Check defaults values are set:

```
spec:
  components:
    cheServer:
      debug: false
      logLevel: INFO
    database:
      credentialsSecretName: postgres-credentials
      externalDb: false
      postgresDb: dbche
      postgresHostName: postgres
      postgresPort: "5432"
      pvc:
        claimSize: 1Gi
    metrics:
      enable: true
  containerRegistry: {}
  devEnvironments:
    defaultNamespace:
      template: <username>-che
    storage:
      pvcStrategy: common
  networking:
    auth:
      gateway:
        configLabels:
          app: che
          component: che-gateway-config

```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
